### PR TITLE
Guards against invalid default pool configuration

### LINKED ISSUE #813: Require default pool config if there are pools in the database
If there are pools in the database, we should require on startup that there's a default pool configured.

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

- moving the seeding of pools to before starting cook scheduler, so that they're already there at startup
- making the pool-seeding setup the database schema
- adding guards on cook scheduler startup against:
  - pools in the database, but no default pool configured
  - the configured default pool does not match any pool in the database

## Why are we making these changes?

We want to prevent the situation where there are pools in the database, but we don't have a valid configuration for which of those pools is the default. Resolves #813.

### CODE REVIEW

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -1632,6 +1632,7 @@ def query():
 
     def test_retrieve_jobs_with_deprecated_api(self):
         pools, _ = util.pools(self.cook_url)
```
Comment (shamsimam) [inline integration/tests/cook/test_basic.py]: How about naming them to `all_pools` and `active_pools`?

File: scheduler/test/cook/test/mesos/pool.clj [hunk 2]
```diff
@@ -0,0 +1,22 @@
+(ns cook.test.mesos.pool
+  (:require [clojure.test :refer :all]
+            [cook.config :as config]
+            [cook.mesos.pool :as pool])
+  (:import (clojure.lang ExceptionInfo)))
+
+(deftest test-guard-invalid-default-pool
```
Comment (shamsimam) [inline scheduler/test/cook/test/mesos/pool.clj]: Can you regex match on the message strings of the exceptions to verify the expected path was triggered?

File: integration/tests/cook/test_basic.py (hunk 1 above)
Comment (dposada) [reply integration/tests/cook/test_basic.py]: Done

File: scheduler/test/cook/test/mesos/pool.clj (hunk 2 above)
Comment (dposada) [reply scheduler/test/cook/test/mesos/pool.clj]: Done

**Comment by shamsimam** [comment]
Awaiting green build

**Comment by dposada** [comment]
@shamsimam Build is now green

**Pull request merged**